tests/commands/test_set.py

```diff
@@ -339,81 +339,63 @@ def test_set_dryrun_flag(repo: Repo, set_values: list[str]) -> None:
     repo.fsck()
 
 
-@pytest.mark.repo_files("laptop_macbook_pro.0",
-                        "dir1/laptop_macbook_pro.1",
-                        "dir1/dir2/laptop_macbook_pro.2",
-                        "dir1/dir2/dir3/laptop_macbook_pro.3",
-                        "dir1/dir2/dir3/dir4/laptop_macbook_pro.4",
-                        "dir1/dir2/dir3/dir4/dir5/laptop_macbook_pro.5",
-                        "dir1/dir2/dir3/dir4/dir5/dir6/laptop_macbook_pro.6",)
+@pytest.mark.repo_files(
+    "laptop_macbook_pro.0",
+    "dir1/laptop_macbook_pro.1",
+    "dir1/dir2/laptop_macbook_pro.2",
+    "dir1/dir2/dir3/laptop_macbook_pro.3",
+    "dir1/dir2/dir3/dir4/laptop_macbook_pro.4",
+    "dir1/dir2/dir3/dir4/dir5/laptop_macbook_pro.5",
+    "dir1/dir2/dir3/dir4/dir5/dir6/laptop_macbook_pro.6",)
 @pytest.mark.parametrize('set_values', values)
-def test_set_depth_flag(repo: Repo, set_values: list[str]) -> None:
+@pytest.mark.parametrize('depth,expected', [
+    ('0', 7), ('1', 1), ('3', 3), ('6', 6), ('10', 7)
+])
+def test_set_depth_flag(
+        repo: Repo, set_values: list[str], depth: str, expected: int) -> None:
     """
     Test correct behavior for `onyo set --depth N KEY=VALUE <assets>` for
-    different values for `--depth N`:
-    - correct error for values smaller then 0
-    - changing correct just assets in same dir for --depth 0
-    - changing assets until sub-dir is --depth N deep
-    - changing all assets if --depth is deeper then deepest sub-directory
-      without error (e.g. deepest folder is 6, but --depth 8 is called)
-    """
-    ret = subprocess.run(['onyo', 'set', '--depth', '-1', '--keys', *set_values], capture_output=True, text=True)
-    # verify output for invalid --depth
-    assert not ret.stdout
-    assert "depth values must be positive, but is -1" in ret.stderr
-    assert ret.returncode == 1
-    repo.fsck()
-
-    ret = subprocess.run(['onyo', 'set', '--depth', '0', '--keys', *set_values], input='n', capture_output=True, text=True)
-    # verify output for --depth 0
-    assert "laptop_macbook_pro.0" in ret.stdout
-    assert "dir1/laptop_macbook_pro.1" not in ret.stdout
-    assert "depth values must be positive" not in ret.stderr
-    assert ret.returncode == 0
-    repo.fsck()
+    different values for `--depth N`.
 
-    ret = subprocess.run(['onyo', 'set', '--depth', '1', '--keys', *set_values], input='n', capture_output=True, text=True)
-    # verify output for --depth 1
-    assert "laptop_macbook_pro.0" in ret.stdout
-    assert "dir1/laptop_macbook_pro.1" in ret.stdout
-    assert "dir1/dir2/laptop_macbook_pro.2" not in ret.stdout
-    assert "depth values must be positive" not in ret.stderr
+    Output is formatted to grab only the assets to be set, which ignores the
+    first line, and then grabs the 2nd line and every 3rd line thereafter. The
```

Comment:
Why the second and third line?

This approach feels brittle. Perhaps counting the number of lines that contain an asset?

Reading further down, each `output` is already looped over. Perhaps loop over each line, see if it contains an asset, test the depth, and count. Then the last check is for the total count (rather than it being the first check).